### ISSUE TITLE
Docker Image Contexts unavailable

### ISSUE THREAD

**Issue by shina2022** [telekom-security/tpotce#1634]
Thank you for the effort and time deployed for the implementation of this awesome security research platform. I installed Tpot on July 18 and it was working perfectly with all the interfaces accessible until today. I started the server but the services were not accessible. 
Having checked all the variables, the only challenge I saw is that the docker exited with error code 18. How can I resolve this please? 

[
![Tpot](https://github.com/user-attachments/assets/7b7ad930-fc4c-4f14-8800-ccc6d14d7614)
](URL)

- What OS are you T-Pot running on?
 Ubuntu 24.04
- What is the version of the OS `lsb_release -a` and `uname -a`?
Ubuntu 24.04 LTS
noble
- What T-Pot version are you currently using (only **T-Pot 24.04.x** is currently supported)?
Tpot version - 24.04
- What architecture are you running on (i.e. hardware, cloud, VM, etc.)?
- XCP-ng Virtualization
- How long has your installation been running?
The installation has been running since July 18, 2024

-Did you install upgrades, packages or use the update script?
I upgraded the Ubuntu package after trying out various methods.

-Did you modify any scripts or configs? 
NO

- Please provide a screenshot of `htop` and `docker stats`.
- 
<img width="941" alt="image" src="https://github.com/user-attachments/assets/de02ffe2-3c2d-4b52-9c6b-9a6c31bf2dc6">

<img width="551" alt="image" src="https://github.com/user-attachments/assets/05ba408d-c7c0-4eea-8055-154bc2bf73ea">

- How much free disk space is available (`df -h`)?
<img width="412" alt="image" src="https://github.com/user-attachments/assets/266b5f30-5c19-4285-b447-641da0b5f99b">
 
- What is the current container status (`dps`)?
<img width="412" alt="image" src="https://github.com/user-attachments/assets/896300fb-9f80-49d1-8e98-790098510fdb">

- On Linux: What is the status of the T-Pot service (`systemctl status tpot`)?
<img width="692" alt="image" src="https://github.com/user-attachments/assets/c7824b57-3404-4549-9c2d-90a0a37da4f1">

- What ports are being occupied? Stop T-Pot `systemctl stop tpot` and run `grc netstat -tulpen`
- 
<img width="746" alt="image" src="https://github.com/user-attachments/assets/1d34a325-727c-4693-b521-4cf81cacd843">

<img width="478" alt="image" src="https://github.com/user-attachments/assets/719d7488-ed15-4932-93d0-f9a905f015e2">

**Comment by t3chn0m4g3** [comment]
What is your docker version?
The contexts are being cancelled either Docker Hub is unavailable or this is related to #1623.
Adjust the pull policy to never in the .env to check if this is related to a Docker Hub problem.

**Comment by shina2022** [comment]
> What is your docker version? The contexts are being cancelled either Docker Hub is unavailable or this is related to #1623. Adjust the pull policy to never in the .env to check if this is related to a Docker Hub problem.

The Docker version is 2.29 and the attempt based on #1623 to install the version 2.28 of noble from Ubuntu 24.04 was not successful. I sincerely appreciate the timely feedback.

**Comment by t3chn0m4g3** [comment]
Check the logs of T-Pot service with `sudo journalctl -u tpot` and look for errors.

Also run `docker pull dtagdevsec/tpotinit:24.04`.

Both will help you to diagnose why the contexts are being cancelled.

Unfortunately I cannot reproduce this.

**Comment by shina2022** [comment]
Thank you @t3chn0m4g3 I tried a fresh install to review the whole process but still had the same challenge. The screenshot is below. This time I adopted Ubuntu 22.04. 

<img width="815" alt="image" src="https://github.com/user-attachments/assets/f0ee4439-7ef9-4b94-9e21-d65f5af42713">

**Comment by t3chn0m4g3** [comment]
The error `context deadline exceeded` points to a networking issue of your setup, nothing I can really help you with.

Ubuntu 22.04 however is not supported.

**Comment by shina2022** [comment]
> The error `context deadline exceeded` points to a networking issue of your setup, nothing I can really help you with.
> 
> Ubuntu 22.04 however is not supported.

Thank you for the update. I launched the earlier installation this morning and the dashboards are now accessible. I suspect the challenge was with the connection to docker-compose. However, I noticed that the logs are no longer accessible as the current attacks captured are what is available in the logs. 

Thank you for your timely feedbacks. It is highly appreciated. I keep learning.
